File: benchmark/template/programs/kotlin/adversarial/ADV-15.kt
fun main() {
    println("ADV-START")
    System.out.flush()
    val xs = mutableListOf<Long>(1, 2, 3, 4, 5)
    var iters = 0
    for (x in xs) {
        iters += 1
        if (x == 2L) xs.add(99L)
    }
    println("OBS=ITERS:" + iters + "|LEN:" + xs.size)
    System.out.flush()
    println("ADV-END")
    System.out.flush()
}

File: benchmark/template/programs/rust/adversarial/ADV-15.rs
use std::io::Write;

fn main() {
    let mut out = std::io::stdout();
    writeln!(out, "ADV-START").unwrap();
    out.flush().unwrap();

    let mut xs: Vec<i64> = vec![1, 2, 3, 4, 5];
    let mut iters: i64 = 0;
    for x in &xs {
        iters += 1;
        if *x == 2 {
            xs.push(99);
        }
    }

    writeln!(out, "OBS=ITERS:{}|LEN:{}", iters, xs.len()).unwrap();
    out.flush().unwrap();
    writeln!(out, "ADV-END").unwrap();
    out.flush().unwrap();
}
